packages/dx-react-grid/docs/guides/editing-in-detail-row.md

```diff
@@ -0,0 +1,232 @@
+# React Grid - Editing in Detail Row
+
+The [detail row](./detail-row.md) is an expandable pane that displays data row details. The detail row does not allow users to edit data out of the box, but you can create a custom plugin to implement this functionality. Before you start, peruse the following help topics:
+
+- [Core Concepts](../../../core/docs/guides/fundamentals.md)
+- [Custom Plugin Development](./custom-plugin-development.md)
+
+The implementation consists of the following steps:
+
+1. [Add a Custom Plugin](#add-a-custom-plugin)
+1. [Handle the Detail Row Expand and Collapse Events](#handle-the-detail-row-expand-and-collapse-events)
+1. [Handle Data Edits](#handle-data-edits)
+1. [Add the Edit Form](#add-the-edit-form)
+1. [Replace the Expand Icon](#replace-the-expand-icon)
+
+## Add a Custom Plugin
+
+For information on custom plugins, refer to the following help topic: [Custom Plugin Development](./custom-plugin-development.md). In the following code, the plugin is called `DetailEditCell`, but you can rename it if required.
+
+```jsx
+const DetailEditCell = () => (
+  <Plugin name="DetailEditCell">
+  </Plugin>
+)
+```
+
+## Handle the Detail Row Expand and Collapse Events
+
+When a user expands or collapses a detail row, the [TableRowDetail](../reference/table-row-detail.md) plugin executes the `toggleDetailRowExpanded` action. Override this action to switch an expanded row to the edit state or switch a collapsed row back to the normal state.
+
+The IDs of expanded rows are included in the `expandedDetailRowIds` array. To find whether a row is expanded or collapsed, check if this array includes the row ID.
+
+```jsx
+const DetailEditCell = () => (
+  <Plugin name="DetailEditCell">
+    <Action
+      name="toggleDetailRowExpanded"
+      action={({ rowId }, { expandedDetailRowIds }, { startEditRows, stopEditRows }) => {
+        const rowIds = [rowId];
+        const isCollapsing = expandedDetailRowIds.indexOf(rowId) > -1;
+        if (isCollapsing) {
+          stopEditRows({ rowIds });
+        } else {
+          startEditRows({ rowIds });
+        }
+      }}
+    />
+  </Plugin>
+)
+```
+
+## Handle Data Edits
+
+To handle data edits, add the following functions:
+
+- `processValueChange` - processes user input
+- `applyChanges` - applies data edits
+- `cancelChanges` - cancels data edits
+
+The detail row and an edit form in it will use these functions later. Implement the functions in the `tableCell` template and pass them as `params` to the [TemplatePlaceholder](../../../dx-react-core/docs/reference/template-placeholder.md) along with the row being edited.
```

Comment:
```suggestion
The detail row and the edit form in it will use these functions later. Implement the functions in the `tableCell` template and pass them as `params` to the [TemplatePlaceholder](../../../dx-react-core/docs/reference/template-placeholder.md) along with the row being edited.
```